### DISCUSSION TITLE
How to update old verify steps with a new required column?

### DISCUSSION THREAD

**Discussion by singpolyma** [theory/sqitch#624] [Q&A]
I have added a change that does an `ALTER TABLE` to add a new `NOT NULL` column with not default value, a new column that must have a value specified when doing an `INSERT`.

I have an older change that does `INSERT` to this table as part of its verify (it's a trigger and testing the trigger runs correctly on `INSERT`) and so now that verify fails.  If I edit it to use the new column, however, then it will pass on the production db but fail during a deploy on an empty db.

What is the best way to handle this?

**Comment by theory** [comment]
This is part of the nature of verify scripts, for better or for worse. The trick is to not use `sqitch verify` on a fully-deployed database, but instead to set `deploy.verify true` in the config so that it verifies each change _immediate after it is deployed_, and before the next change is deployed. This will ensure that older change's verify script only operates on the table before the `NOT NULL` constraint is added. A separate verify for the new change that adds the constraint should verify the constraint, of course.
